cmd/start.go

```diff
@@ -97,29 +89,18 @@ func init() {
 	setDefault(OSMOSIS_BLOCKS, "20000")
 	setDefault(PRICES_SUBJECT, "syntropy_defi.price.single.OSMO")
 
-	flagPublisherName = startCmd.Flags().String("publisher-name", os.Getenv(OSMOSIS_NAME), "NATS publisher name as in {prefix}.{name}.>")
-	flagTendermintAPI = startCmd.Flags().String("tendermint-api", os.Getenv(OSMOSIS_TENDERMINT), "Full address to the Tendermint RPC")
-	flagRPCAPI = startCmd.Flags().String("app-api", os.Getenv(OSMOSIS_RPC), "Full address to the Applications RPC")
-	flagGRPCAPI = startCmd.Flags().String("grpc-api", os.Getenv(OSMOSIS_GRPC), "Full address to the Applications gRPC")
-
-	flagPricesSubject = startCmd.Flags().String("prices-subject", os.Getenv(PRICES_SUBJECT), "Subject for prices feed to subscribe to")
-
-	pools := SplitAndTrimEmpty(os.Getenv(OSMOSIS_POOLS), ",", " \t\r\n\b")
-	dp := make([]int64, len(pools))
-	for i, p := range pools {
-		val, err := strconv.ParseInt(p, 10, 64)
-		if err != nil {
-			log.Fatal(err)
-		}
-		dp[i] = val
-	}
-	startCmd.Flags().Int64Slice("pool-ids", dp, "A list of Osmosis pools to stream volume and liquidity each block")
+	f := startCmd.Flags()
+	flagPublisherName = f.String("publisher-name", os.Getenv(OSMOSIS_NAME), "NATS publisher name as in {prefix}.{name}.>")
+	flagTendermintAPI = f.String("tendermint-api", os.Getenv(OSMOSIS_TENDERMINT), "Full address to the Tendermint RPC")
+	flagRPCAPI = f.String("app-api", os.Getenv(OSMOSIS_RPC), "Full address to the Applications RPC")
+	flagGRPCAPI = f.String("grpc-api", os.Getenv(OSMOSIS_GRPC), "Full address to the Applications gRPC")
+	flagPricesSubject = f.String("prices-subject", os.Getenv(PRICES_SUBJECT), "Subject for prices feed to subscribe to")
+	flagPoolIds = f.VarPF(NewPoolIds(os.Getenv(OSMOSIS_POOLS)), "pool-ids", "", "A list of Osmosis pools to stream volume and liquidity each block").Value.(*PoolIds)
 
-	envBlocks := os.Getenv(OSMOSIS_BLOCKS)
-	blocks, err := strconv.ParseUint(envBlocks, 10, 64)
+	blocks, err := strconv.ParseUint(os.Getenv(OSMOSIS_BLOCKS), 10, 64)
 	if err != nil {
 		blocks = 17000
```

Comment:
Would be great to fix this default to the same value as `OSMOSIS_BLOCKS` env. variable :)

Reply:
Sure, will do! thanks

Reply:
No need to set it to OSMOSIS_BLOCKS value. Just set it to `20000` as an integer instead of `17000`.

Reply:
oki done